### ISSUE TITLE
[BUG] Spamtraps are not always handled as such

### ISSUE THREAD

**Issue by ghost** [rspamd/rspamd#3592]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Read about bug reporting in general: https://rspamd.com/doc/faq.html#how-to-report-bugs-found-in-rspamd
    * Enabled relevant debugging logs: https://rspamd.com/doc/faq.html#how-to-debug-some-module-in-rspamd 
    * Checked the FAQs about Core files in case of fatal crash: https://rspamd.com/doc/faq.html#how-to-figure-out-why-rspamd-process-crashed
    * Tried ASAN package and obtained the ASAN report (if possible): https://rspamd.com/doc/faq.html#asan-builds 
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+user%3Arspamd
    * Checked that there is not already an experimental package or master branch

**Describe the bug**
`rspamd` supports setting up [spamtraps](https://rspamd.com/doc/modules/spamtrap.html), and a list of addresses or domains which should be treated as such can be supplied via a file.

However, in some cases, `rspamd` does not seem to detect a spamtrap as such, causing a message to be rejected (which is bad for obvious reasons), while handling another message to another spamtrap at the very same time correctly. This does not seem to be a race condition between two threads or modules, as sometimes, a spamtrap is detected as such, while hours later, it is not.

Log samples are provided below.

### Steps to Reproduce

1. Set up `rspamd` and a MTA to work properly.
2. Set up some spamtraps and configure them according to the [spamtrap module documentation](https://rspamd.com/doc/modules/spamtrap.html).
3. Wait for spam message to reach the spamtraps and look out for rejected messages.

**Expected behavior**
Especially if `action` is set to `accept`, my expectation of configured spamtraps would be to accept any incoming message at any time. If the is no passthrough action set, messages should be at least tagged (`SPAMTRAP` by default) reliably if they were addressed to a spamtrap.

### Versions

```
[user@machine ~]$ rspamd --version
Rspamd daemon version 2.6
```
```
[user@machine ~]$ cat /etc/os-release 
NAME=FreeBSD
VERSION=12.2--HBSD
VERSION_ID=12.2
ID=freebsd
ANSI_COLOR="0;31"
PRETTY_NAME="FreeBSD 12.2--HBSD"
CPE_NAME=cpe:/o:freebsd:freebsd:12.2
HOME_URL=https://FreeBSD.org/
BUG_REPORT_URL=https://bugs.FreeBSD.org/
```

### Additional Information
Message sent to spamtrap but was rejected:
```
2020-12-29 15:07:43 #85048(rspamd_proxy) <1b7f66>; proxy; proxy_accept_socket: accepted milter connection from /var/spool/postfix/public/rspamd_proxy.sock port 0
2020-12-29 15:07:44 #85048(rspamd_proxy) <1b7f66>; milter; rspamd_milter_process_command: got connection from 85.xxx.xxx.xxx:x
2020-12-29 15:07:44 #85048(rspamd_proxy) <1b7f66>; proxy; rspamd_message_parse: loaded message; id: <x@[SPAMMER DOMAIN].com>; queue-id: <x>; size: 33676; checksum: <x>
2020-12-29 15:07:44 #85048(rspamd_proxy) <1b7f66>; proxy; rspamd_mime_part_detect_language: detected part language: en
2020-12-29 15:07:44 #85048(rspamd_proxy) <1b7f66>; proxy; rspamd_mime_part_detect_language: detected part language: en
2020-12-29 15:07:44 #85048(rspamd_proxy) <1b7f66>; proxy; rspamd_spf_maybe_return: stored record for [SPAMMER DOMAIN].com (x) in LRU cache for 1844 seconds, 13/2000 elements in the cache
2020-12-29 15:07:44 #85048(rspamd_proxy) <1b7f66>; proxy; dkim_module_key_handler: stored DKIM key for default._domainkey.[SPAMMER DOMAIN].com in LRU cache for 38400 seconds, 15/2000 elements in the cache
2020-12-29 15:07:44 #85048(rspamd_proxy) <1b7f66>; proxy; rspamd_add_passthrough_result: <x@[SPAMMER DOMAIN].com>: set pre-result to 'reject' (no score): 'Rejected due to policy violation: Contains blacklisted URL.' from force_actions(1)
2020-12-29 15:07:44 #85048(rspamd_proxy) <1b7f66>; lua; dmarc.lua:532: <x@[SPAMMER DOMAIN].com> dmarc report saved for [SPAMMER DOMAIN].com
2020-12-29 15:07:45 #85048(rspamd_proxy) <1b7f66>; proxy; rspamd_stat_check_autolearn: <x@[SPAMMER DOMAIN].com>: autolearn spam for classifier 'bayes' as message's action is reject, score: 15.99
2020-12-29 15:07:45 #85048(rspamd_proxy) <1b7f66>; proxy; rspamd_task_process: learn error: Cannot learn statistics when passthrough result has been set; not classified
2020-12-29 15:07:45 #85048(rspamd_proxy) <1b7f66>; proxy; rspamd_task_write_log: id: <x@[SPAMMER DOMAIN].com>, qid: <x>, ip: 85.xxx.xxx.xxx, from: <info@[SPAMMER DOMAIN].com>, (default: T (reject): [15.99/11.00] [URIBL_SBL_CSS(6.50){85.xxx.xxx.xxx:[SPAMMER DOMAIN].com:url;},DBL_SPAM(6.00){[SPAMMER DOMAIN].com:dkim;},RBL_SPAMHAUS_CSS(2.00){85.xxx.xxx.xxx:from;},SUBJECT_HAS_CURRENCY(1.00){},URI_COUNT_ODD(1.00){29;},DMARC_POLICY_ALLOW(-0.50){[SPAMMER DOMAIN].com;reject;},HTML_SHORT_LINK_IMG_3(0.50){},R_DKIM_ALLOW(-0.20){[SPAMMER DOMAIN].com:s=default;},R_SPF_ALLOW(-0.20){+ip4:85.xxx.xxx.xxx/27;},MIME_GOOD(-0.10){multipart/alternative;text/plain;},HAS_LIST_UNSUB(-0.01){},ARC_NA(0.00){},ARC_SIGNED(0.00){x:s=x:i=1;},ASN(0.00){asn:x, ipnet:85.xxx.xxx.xxx, country:RO;},DKIM_TRACE(0.00){[SPAMMER DOMAIN].com:+;},FORCE_ACTION_CUSTOM_FORCEACTION_URLS_RESOLVE_BLACKLISTED(0.00){reject;},FROM_EQ_ENVFROM(0.00){},FROM_HAS_DN(0.00){},HAS_REPLYTO(0.00){info@[SPAMMER DOMAIN].com;},MID_RHS_MATCH_FROM(0.00){},MIME_TRACE(0.00){0:+;1:+;2:~;},PRECEDENCE_BULK(0.00){},RCPT_COUNT_ONE(0.00){1;},RCVD_COUNT_ZERO(0.00){0;},REPLYTO_EQ_FROM(0.00){},SUBJECT_ENDS_EXCLAIM(0.00){},SUBJECT_HAS_QUESTION(0.00){},TO_DN_EQ_ADDR_ALL(0.00){},TO_MATCH_ENVRCPT_ALL(0.00){}]), len: 33676, time: 1147.747ms, dns req: 49, digest: <x>, rcpts: <x>, mime_rcpts: <x>, forced: reject "Rejected due to policy violation: Contains blacklisted URL."; score=nan (set by force_actions)
2020-12-29 15:07:45 #85048(rspamd_proxy) <1b7f66>; proxy; rspamd_protocol_http_reply: regexp statistics: 85 pcre regexps scanned, 9 regexps matched, 187 regexps total, 13 regexps cached, 102.99KiB scanned using pcre, 102.99KiB scanned total
```
Message sent to a different spamtrap at the very same time, accepted and tagged correctly:
```
2020-12-29 15:07:43 #85048(rspamd_proxy) <4c523b>; proxy; proxy_accept_socket: accepted milter connection from /var/spool/postfix/public/rspamd_proxy.sock port 0
2020-12-29 15:07:44 #85048(rspamd_proxy) <4c523b>; milter; rspamd_milter_process_command: got connection from 85.xxx.xxx.xxx:x
2020-12-29 15:07:44 #85048(rspamd_proxy) <4c523b>; proxy; rspamd_message_parse: loaded message; id: <x@[SPAMMER DOMAIN].com>; queue-id: <x>; size: 33664; checksum: <x>
2020-12-29 15:07:44 #85048(rspamd_proxy) <4c523b>; proxy; rspamd_mime_part_detect_language: detected part language: en
2020-12-29 15:07:44 #85048(rspamd_proxy) <4c523b>; proxy; rspamd_mime_part_detect_language: detected part language: en
2020-12-29 15:07:44 #85048(rspamd_proxy) <4c523b>; proxy; rspamd_spf_maybe_return: stored record for [SPAMMER DOMAIN].com (x) in LRU cache for 1844 seconds, 13/2000 elements in the cache
2020-12-29 15:07:44 #85048(rspamd_proxy) <4c523b>; proxy; dkim_module_key_handler: stored DKIM key for default._domainkey.[SPAMMER DOMAIN].com in LRU cache for 38400 seconds, 15/2000 elements in the cache
2020-12-29 15:07:44 #85048(rspamd_proxy) <4c523b>; lua; dmarc.lua:532: <x@[SPAMMER DOMAIN].com> dmarc report saved for [SPAMMER DOMAIN].com
2020-12-29 15:07:44 #85048(rspamd_proxy) <4c523b>; proxy; rspamd_symcache_finalize_item: slow rule: MSBL_EBL(283): 300.75 ms; enable slow timer delay
2020-12-29 15:07:45 #85048(rspamd_proxy) <4c523b>; proxy; rspamd_symcache_finalize_item: slow rule: RCVD_IN_DNSWL(298): 779.27 ms; enable slow timer delay
2020-12-29 15:07:45 #85048(rspamd_proxy) <4c523b>; proxy; rspamd_symcache_finalize_item: slow rule: RBL_SENDERSCORE(294): 400.38 ms; enable slow timer delay
2020-12-29 15:07:45 #85048(rspamd_proxy) <4c523b>; bayes; bayes_classify: no tokens found in bayes database (1068 total tokens, 1036 text tokens), ignore stats
2020-12-29 15:07:45 #85048(rspamd_proxy) <4c523b>; proxy; fuzzy_check_lua_process_learn: no fuzzy rules found for flag 1
2020-12-29 15:07:45 #85048(rspamd_proxy) <4c523b>; lua; spamtrap.lua:64: spamtrap found: <x>
2020-12-29 15:07:45 #85048(rspamd_proxy) <4c523b>; proxy; rspamd_add_passthrough_result: <x@[SPAMMER DOMAIN].com>: set pre-result to 'no action' (no score): 'unknown error' from spamtrap(1)
2020-12-29 15:07:45 #85048(rspamd_proxy) <4c523b>; proxy; rspamd_task_process: learn error: Cannot learn statistics when passthrough result has been set; not classified
2020-12-29 15:07:45 #85048(rspamd_proxy) <4c523b>; proxy; rspamd_task_write_log: id: <x@[SPAMMER DOMAIN].com>, qid: <x>, ip: 85.xxx.xxx.xxx, from: <info@[SPAMMER DOMAIN].com>, (default: F (no action): [22.99/11.00] [DBL_SPAM(6.50){[SPAMMER DOMAIN].com:dkim;},SPAMTRAP(6.00){x;},URIBL_SBL_CSS(6.00){85.xxx.xxx.xxx:[SPAMMER DOMAIN].com:url;},RBL_SPAMHAUS_CSS(2.00){85.xxx.xxx.xxx:from;},SUBJECT_HAS_CURRENCY(1.00){},URI_COUNT_ODD(1.00){29;},HTML_SHORT_LINK_IMG_3(0.50){},BAD_REP_POLICIES(0.10){},MIME_GOOD(-0.10){multipart/alternative;text/plain;},HAS_LIST_UNSUB(-0.01){},ARC_NA(0.00){},ARC_SIGNED(0.00){x:s=x:i=1;},ASN(0.00){asn:x, ipnet:85.xxx.xxx.xxx, country:RO;},DKIM_TRACE(0.00){[SPAMMER DOMAIN].com:+;},DMARC_POLICY_ALLOW(0.00){[SPAMMER DOMAIN].com;reject;},FROM_EQ_ENVFROM(0.00){},FROM_HAS_DN(0.00){},HAS_REPLYTO(0.00){info@[SPAMMER DOMAIN].com;},MID_RHS_MATCH_FROM(0.00){},MIME_TRACE(0.00){0:+;1:+;2:~;},PRECEDENCE_BULK(0.00){},RCPT_COUNT_ONE(0.00){1;},RCVD_COUNT_ZERO(0.00){0;},REPLYTO_EQ_FROM(0.00){},R_DKIM_ALLOW(0.00){[SPAMMER DOMAIN].com:s=default;},R_SPF_ALLOW(0.00){+ip4:85.xxx.xxx.xxx/27;},SUBJECT_ENDS_EXCLAIM(0.00){},SUBJECT_HAS_QUESTION(0.00){},TO_DN_EQ_ADDR_ALL(0.00){},TO_MATCH_ENVRCPT_ALL(0.00){}]), len: 33664, time: 1597.438ms, dns req: 49, digest: <x>, rcpts: <x>, mime_rcpts: <x>, forced: no action "unknown error"; score=nan (set by spamtrap)
2020-12-29 15:07:45 #85048(rspamd_proxy) <4c523b>; proxy; rspamd_protocol_http_reply: regexp statistics: 85 pcre regexps scanned, 9 regexps matched, 187 regexps total, 13 regexps cached, 102.91KiB scanned using pcre, 102.91KiB scanned total
```
Third message rejected, suggesting this is not (entirely) a race condition:
```
2020-12-23 17:45:29 #23585(rspamd_proxy) <6b3d14>; proxy; proxy_accept_socket: accepted milter connection from /var/spool/postfix/public/rspamd_proxy.sock port 0
2020-12-23 17:45:30 #23585(rspamd_proxy) <6b3d14>; milter; rspamd_milter_process_command: got connection from 167.xxx.xxx.xxx:x
2020-12-23 17:45:30 #23585(rspamd_proxy) <6b3d14>; proxy; rspamd_message_parse: loaded message; id: <x@[SPAMMER DOMAIN].com>; queue-id: <x>; size: 12346; checksum: <x>
2020-12-23 17:45:30 #23585(rspamd_proxy) <6b3d14>; proxy; rspamd_mime_part_detect_language: detected part language: en
2020-12-23 17:45:30 #23585(rspamd_proxy) <6b3d14>; proxy; dkim_module_key_handler: stored DKIM key for s1._domainkey.n.[SPAMMER DOMAIN].com in LRU cache for 1800 seconds, 34/2000 elements in the cache
2020-12-23 17:45:30 #23585(rspamd_proxy) <6b3d14>; proxy; rspamd_symcache_finalize_item: slow rule: DKIM_CHECK(97): 300.66 ms; enable slow timer delay
2020-12-23 17:45:30 #23585(rspamd_proxy) <6b3d14>; proxy; rspamd_add_passthrough_result: <x@[SPAMMER DOMAIN].com>: set pre-result to 'reject' (no score): 'Rejected due to policy violation: Client IP address blacklisted.' from force_actions(1)
2020-12-23 17:45:30 #23585(rspamd_proxy) <6b3d14>; proxy; rspamd_spf_maybe_return: stored record for mail.n.[SPAMMER DOMAIN].com (x) in LRU cache for 1800 seconds, 28/2000 elements in the cache
2020-12-23 17:45:30 #23585(rspamd_proxy) <6b3d14>; proxy; rspamd_symcache_finalize_item: slow rule: SPF_CHECK(313): 368.55 ms
2020-12-23 17:45:30 #23585(rspamd_proxy) <6b3d14>; proxy; rspamd_symcache_finalize_item: slow rule: RBL_SENDERSCORE(294): 397.40 ms; enable slow timer delay
2020-12-23 17:45:31 #23585(rspamd_proxy) <6b3d14>; proxy; rspamd_symcache_finalize_item: slow rule: HFILTER(252): 655.49 ms
2020-12-23 17:45:31 #23585(rspamd_proxy) <6b3d14>; proxy; rspamd_stat_check_autolearn: <x@[SPAMMER DOMAIN].com>: autolearn spam for classifier 'bayes' as message's action is reject, score: 9.47
2020-12-23 17:45:31 #23585(rspamd_proxy) <6b3d14>; proxy; rspamd_task_process: learn error: Cannot learn statistics when passthrough result has been set; not classified
2020-12-23 17:45:31 #23585(rspamd_proxy) <6b3d14>; proxy; rspamd_task_write_log: id: <x@[SPAMMER DOMAIN].com>, qid: <x>, ip: 167.xxx.xxx.xxx, from: <bounces@mail.n.[SPAMMER DOMAIN].com>, (default: T (reject): [9.47/11.00] [RBL_NIXSPAM(4.00){167.xxx.xxx.xxx:from;},HTML_SHORT_LINK_IMG_1(2.00){},HFILTER_URL_ONLY(1.37){0.62365591397849;},URI_COUNT_ODD(1.00){25;},R_PARTS_DIFFER(0.82){91.4%;},MV_CASE(0.50){},FORGED_SENDER(0.30){admin@[ALSO SPAMMER DOMAIN].com;bounces@mail.n.[SPAMMER DOMAIN].com;},R_DKIM_ALLOW(-0.20){n.[SPAMMER DOMAIN].com:s=s1;},R_SPF_ALLOW(-0.20){+ip4:167.xxx.xxx.xxx/17;},MIME_GOOD(-0.10){multipart/alternative;text/plain;},IP_REPUTATION_HAM(-0.01){asn: x(0.00), country: US(-0.01), ip: 167.xxx.xxx.xxx(0.00);},HAS_LIST_UNSUB(-0.01){},ASN(0.00){asn:x, ipnet:167.xxx.xxx.xxx/18, country:US;},DKIM_TRACE(0.00){n.[SPAMMER DOMAIN].com:+;},FORCE_ACTION_CUSTOM_FORCEACTION_RBL_BLACKLISTED(0.00){reject;},FROM_HAS_DN(0.00){},FROM_NEQ_ENVFROM(0.00){admin@[ALSO SPAMMER DOMAIN].com;bounces@mail.n.[SPAMMER DOMAIN].com;},MIME_TRACE(0.00){0:+;1:+;2:~;},PRECEDENCE_BULK(0.00){},RCPT_COUNT_ONE(0.00){1;},RCVD_COUNT_TWO(0.00){2;},RCVD_TLS_LAST(0.00){},RWL_MAILSPIKE_POSSIBLE(0.00){167.xxx.xxx.xxx:from;},TAGGED_FROM(0.00){8946241-2b7b-x=x.x;},TO_DN_NONE(0.00){},TO_MATCH_ENVRCPT_ALL(0.00){}]), len: 12346, time: 856.712ms, dns req: 38, digest: <x>, rcpts: <x>, mime_rcpts: <x>, forced: reject "Rejected due to policy violation: Client IP address blacklisted."; score=nan (set by force_actions)
2020-12-23 17:45:31 #23585(rspamd_proxy) <6b3d14>; proxy; rspamd_protocol_http_reply: regexp statistics: 153 pcre regexps scanned, 2 regexps matched, 187 regexps total, 11 regexps cached, 49.21KiB scanned using pcre, 49.21KiB scanned total
```
Spamtrap configuration is as follows:
```
spamtrap {
    action = "accept";
    score = 6;
    enabled = true;
    map = "file:///usr/local/etc/rspamd/local.d/maps/spamtrap.map";
    learn_spam = true;
}
```

Thank you for having a look at this.

**Comment by moisseev** [comment]
Does it only happen to multi-recipient messages or single-recipient messages affected as well?

**Comment by heraklit256** [comment]
This happened to single-recipient messages only, as I did not observe multi-recipient messages to my spamtraps so far.

**Comment by fatalbanana** [comment]
> forced: reject "Rejected due to policy violation: Contains blacklisted URL."; score=nan (set by force_actions)

The issue is about interaction with your `force_actions` configuration. However, I expect it should work as you do. In testing on a recent build it seems it does indeed work. On version 2.6 I can reproduce your issue. Hence I think this issue is already resolved.

**Comment by heraklit256** [comment]
Thank you for your reply.

Alas, I still observe this behaviour on a Debian machine running `rspamd` 2.7 (destination address is a spamtrap, but was not detected as such):
```
2021-01-25 10:22:34 #21767(rspamd_proxy) <ccbaef>; proxy; proxy_accept_socket: accepted milter connection from /var/run/rspamd/rspamd_proxy.sock port 0
2021-01-25 10:22:38 #21767(rspamd_proxy) <ccbaef>; milter; rspamd_milter_process_command: got connection from 122.192.207.199:59484
2021-01-25 10:22:39 #21767(rspamd_proxy) <ccbaef>; proxy; rspamd_message_parse: loaded message; id: <20210125102238.B664F14E029@vps17833.com>; queue-id: <4DPQsm2cc7z1hJ>; size: 1969; checksum: <5998913a638f13fa2335a37420c6dfa2>
2021-01-25 10:22:39 #21767(rspamd_proxy) <ccbaef>; proxy; rspamd_mime_text_part_utf8_convert: converted text part from ibm-1383_P110-1999 ('GB2312' announced) to UTF-8 inlen: 905, outlen: 906 (904 UTF16 chars)
2021-01-25 10:22:39 #21767(rspamd_proxy) <ccbaef>; proxy; rspamd_mime_part_detect_language: detected part language: en
2021-01-25 10:22:39 #21767(rspamd_proxy) <ccbaef>; proxy; fuzzy_insert_result: found exact fuzzy hash(txt) 6dcb53dc3336381f1a5dfd782145336832acb8b3e125edb4fcc08b9310965960063d741638e662354f00c51644976247e74c07fee212f30b1d94fe76cb237237 with weight: 0.87, probability 1.00, in list: FUZZY_DENIED:1; added on 28.12.2020 02:41:44 GMT
2021-01-25 10:22:41 #21767(rspamd_proxy) <ccbaef>; proxy; rspamd_symcache_finalize_item: slow rule: RCVD_IN_DNSWL(299): 1715.91 ms; enable slow timer delay
2021-01-25 10:22:41 #21767(rspamd_proxy) <ccbaef>; proxy; rspamd_add_passthrough_result: <20210125102238.B664F14E029@vps17833.com>: set pre-result to 'reject' (no score): 'Rejected due to policy violation: Client IP address blacklisted' from force_actions(0)
2021-01-25 10:22:41 #21767(rspamd_proxy) <ccbaef>; proxy; rspamd_symcache_finalize_item: slow rule: RBL_SEM(298): 372.85 ms; enable slow timer delay
2021-01-25 10:22:41 #21767(rspamd_proxy) <ccbaef>; proxy; rspamd_symcache_finalize_item: slow rule: URIBL_MULTI(284): 410.45 ms
2021-01-25 10:22:41 #21767(rspamd_proxy) <ccbaef>; proxy; rspamd_stat_check_autolearn: <20210125102238.B664F14E029@vps17833.com>: autolearn spam for classifier 'bayes' as message's action is reject, score: 32.93
2021-01-25 10:22:41 #21767(rspamd_proxy) <ccbaef>; proxy; rspamd_task_process: learn error: Cannot learn statistics when passthrough result has been set; not classified
2021-01-25 10:22:41 #21767(rspamd_proxy) <ccbaef>; proxy; rspamd_task_write_log: id: <20210125102238.B664F14E029@vps17833.com>, qid: <4DPQsm2cc7z1hJ>, ip: 122.192.207.199, from: <wwwwu@zgvtllmpy.com>, (default: T (reject): [32.93/11.00] [FUZZY_DENIED(10.51){1:6dcb53dc33:1.00:txt;},RBL_SPAMHAUS_XBL(4.00){122.192.207.199:from;},INVALID_POSTFIX_RECEIVED(3.00){},HFILTER_HOSTNAME_UNKNOWN(2.50){},RBL_SENDERSCORE(2.00){122.192.207.199:from;},RBL_SPAMHAUS_CSS(2.00){122.192.207.199:from;},RBL_VIRUSFREE_BOTNET(2.00){122.192.207.199:from;},HFILTER_FROMHOST_NORES_A_OR_MX(1.50){zgvtllmpy.com;},SUBJ_EXCESS_BASE64(1.50){},FAKE_REPLY(1.00){},HFILTER_HELO_IP_A(1.00){vps17833.com;},RDNS_NONE(1.00){},MV_CASE(0.50){},HFILTER_HELO_NORES_A_OR_MX(0.30){vps17833.com;},MIME_BASE64_TEXT(0.10){},MIME_GOOD(-0.10){text/plain;},RCVD_NO_TLS_LAST(0.10){},IP_REPUTATION_SPAM(0.01){asn: 4837(0.00), country: CN(0.01), ip: 122.192.207.199(0.00);},ARC_NA(0.00){},ARC_SIGNED(0.00){x:s=202003rsa:i=1;},ASN(0.00){asn:4837, ipnet:122.192.0.0/14, country:CN;},DMARC_NA(0.00){zgvtllmpy.com;},FORCE_ACTION_CUSTOM_FORCEACTION_RBL_BLACKLISTED(0.00){reject;},FREEMAIL_REPLYTO(0.00){126.com;},FROM_EQ_ENVFROM(0.00){},FROM_HAS_DN(0.00){},HAS_REPLYTO(0.00){xiexian45269741@126.com;},MIME_TRACE(0.00){0:+;},PREVIOUSLY_DELIVERED(0.00){x;},RCPT_COUNT_ONE(0.00){1;},RCVD_COUNT_TWO(0.00){2;},REPLYTO_DOM_NEQ_FROM_DOM(0.00){},R_DKIM_NA(0.00){},R_SPF_NA(0.00){no SPF record;},TO_DN_ALL(0.00){},TO_MATCH_ENVRCPT_ALL(0.00){}]), len: 1969, time: 2328.922ms, dns req: 32, digest: <5998913a638f13fa2335a37420c6dfa2>, rcpts: <x>, mime_rcpts: <x>, forced: reject "Rejected due to policy violation: Client IP address blacklisted"; score=nan (set by force_actions)
2021-01-25 10:22:41 #21767(rspamd_proxy) <ccbaef>; proxy; rspamd_protocol_http_reply: regexp statistics: 0 pcre regexps scanned, 5 regexps matched, 174 regexps total, 84 regexps cached, 0B scanned using pcre, 3.64KiB scanned total
```

No offense intended, but using force actions seems to have some side effects - especially on Bayes, as it seems to ignore messages rejected due to a forced action for a reason I did not get yet, but also on spamtraps. Personally, I use them excessively, since this makes mail delivery deterministic, thus allowing to enforce a documented policy. However, I am getting the feeling that this is not the intended usage of force actions, unless I misunderstood something.

Long story short: This bug still appears using `rspamd` 2.7 - sorry.
